### PR TITLE
Add tests to the weather_cli file

### PR DESCRIPTION
**Pull request by Krisha434**
Basic test suite for the WeatherDashboard CLI, verifying core functionality like data fetching, display, and history management using mocked API responses.

### CODE REVIEW

File: testAPI.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    dashboard = WeatherDashboard()
+    result = dashboard.load_history()
+    assert result == dashboard.history_file
+
+@patch("requests.get")
+@patch.dict(os.environ, {"api_key": "test_key"})
+def test_fetch_weather_success(mock_get):
+    """Test successful weather fetch."""
+    mock_response = type("Response", (), {
+        "status_code": 200,
```
Comment (sreedevijain19) [inline testAPI.py:55]: To include edge cases for invalid  or incomplete API responses (eg, invalid city name, missing data or API errors like 404/500)

**Review by sreedevijain19 — COMMENTED** [review]
Add requirements

**Review by sreedevijain19 — APPROVED** [review]

**Review by sreedevijain19 — COMMENTED** [review]
Can Merge
